Automattic/wp-calypso
# Domain Upsell: Update the My home design

Related to https://github.com/Automattic/dotcom-forge/issues/2207

## Proposed Changes

Update the Domain upsell Design on `/home` to match the Figma.
We also removed the `strike` on the current domain name on `/me` as commented here (pebzTe-RL-p2#comment-1694)
As now `/home` and `/me` have different designs, we create a new component for the profile page.

Figma: LruC8ztBCLJMFNkQN1JNm3-fi-2133_20773

As the PR got big because of the suggestions, I detailed the changes:
- [x] Move `/me` Domain Upsell to its own component
- [x] Remove all `/me` related data from the `/home` Domain Upsell component
- [x] Update styles on the `/home` Domain Upsell
- [x] Update e2e tests with suggestions from [here](https://github.com/Automattic/wp-calypso/pull/76326#discussion_r1178875347)

## Screenshots
| Desktop |  Mobile |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/402286/234735781-b5c4f309-8306-4c42-a516-7fdb0d12f27c.png) | ![image](https://user-images.githubusercontent.com/402286/234735766-9a7730f8-3260-4eaa-8b31-b5c75d01ff0a.png) | 
| ![image](https://user-images.githubusercontent.com/402286/234735785-54e06e06-f4a5-4c4b-8cda-8b23b15909a6.png) | ![image](https://user-images.githubusercontent.com/402286/234735815-d37daf08-005d-451c-a489-b6b31ebc6ee9.png) |

#### `/me` design update
<img width="735" alt="image" src="https://user-images.githubusercontent.com/402286/235012844-b3b3c5f6-35c6-40b8-9853-4212cc4fddf8.png">

## Tests

#### E2e
* First of all run `yarn build` on `/packages/calypso-e2e`
* Run `yarn workspace wp-e2e-tests test -- domain-upsell/domain-upsell__my-home`

#### Component
* For my home, run `yarn run test-client client/my-sites/customer-home/cards/features/domain-upsell/test/index.jsx`
* For profile, run `yarn run test-client client/me/domain-upsell/test/index.jsx`

## Testing Instructions

* Go to /home/:siteSlug
* With a site matching the conditions, you should see the Domain Upsell (email-verified + site without a custom domain + site setup not completed)
* Check the new design (desktop & mobile).
* Check `/me` to check the new design (desktop & mobile)

## Pre-merge Checklist



- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
